src/language.cpp

```diff
@@ -309,24 +300,76 @@ std::string Language::GetText(StringID number)
 
 	if (number < lengthof(default_strings)) return default_strings[number];
 
-	if (number < lengthof(this->registered) && this->registered[number] != nullptr) {
-		const std::string &text = this->registered[number]->GetString();
-		if (text.empty()) return "<empty text>";
-		return text;
+	const char *str = languages[_current_language].GetPlural(number, count);
+	if (str != nullptr) return *str != '\0' ? str : "<empty translation>";
+
+	if (_current_language != SOURCE_LANGUAGE) {
+		str = languages[SOURCE_LANGUAGE].GetPlural(number, count);
+		if (str != nullptr) return *str != '\0' ? str : "<empty string>";
```

Comment:
Rather than sticking the `if (not the source language)` around here, do it with the previous attempt where you pull the translated string.

If I am playing with the `SOURCE_LANGUAGE`, I won't get the `<empty translation>` in that case.